Geometry/CSCGeometryBuilder/src/CSCGeometryParsFromDD.cc

```diff
@@ -301,3 +460,366 @@ bool CSCGeometryParsFromDD::build(const DDCompactView* cview,
+      if (chamberType == rdp.pChamberType[ct]) {
+        break;
+      }
+    }
+
+    if (ct < rdp.pChamberType.size() && rdp.pChamberType[ct] == chamberType) {
+      chSpecsAlreadyExist = true;
+    } else {
+      std::string_view my_name_bis = fv.name();
+      std::string my_name_tris = std::string(my_name_bis);
+      std::vector<std::string_view> namesInPath = mypar.names("//" + my_name_tris);
+      std::string my_string = "ChamberSpecs_";
+      int index = -1;
+      for (vector<string_view>::size_type i = 0; i < namesInPath.size(); ++i) {
+        std::size_t found = namesInPath[i].find(my_string);
+        if (found != std::string::npos)
+          index = i;
+      }
+      uparvals = fv.get<std::vector<double>>(std::string(namesInPath[index]), "upar");
+
+      auto it = uparvals.begin();
+      it = uparvals.insert(it, uparvals.size());
+      auto noofanonparams = fv.get<double>("NoOfAnonParams");
+      noOfAnonParams = static_cast<int>(noofanonparams);
+
+      for (auto i : fv.get<std::vector<double>>(std::string(namesInPath[index]), "NumWiresPerGrp")) {
+        wg.wiresInEachGroup.emplace_back(int(i));
+      }
+
+      for (auto i : fv.get<std::vector<double>>(std::string(namesInPath[index]), "NumGroups")) {
+        wg.consecutiveGroups.emplace_back(int(i));
+      }
+
+      auto wirespacing = fv.get<double>("WireSpacing");
+      wg.wireSpacing = static_cast<double>(wirespacing);
+
+      auto alignmentpintofirstwire = fv.get<double>("AlignmentPinToFirstWire");
+      wg.alignmentPinToFirstWire = static_cast<double>(alignmentpintofirstwire);
+
+      auto totnumwiregroups = fv.get<double>("TotNumWireGroups");
+      wg.numberOfGroups = static_cast<int>(totnumwiregroups);
+
+      auto lengthoffirstwire = fv.get<double>("LengthOfFirstWire");
+      wg.narrowWidthOfWirePlane = static_cast<double>(lengthoffirstwire);
+
+      auto lengthoflastwire = fv.get<double>("LengthOfLastWire");
+      wg.wideWidthOfWirePlane = static_cast<double>(lengthoflastwire);
+
+      auto radialextentofwireplane = fv.get<double>("RadialExtentOfWirePlane");
+      wg.lengthOfWirePlane = static_cast<double>(radialextentofwireplane);
+
+      uparvals.emplace_back((wg.wireSpacing) * 10.0);
+      uparvals.emplace_back(wg.alignmentPinToFirstWire);
+      uparvals.emplace_back(wg.numberOfGroups);
+      uparvals.emplace_back(wg.narrowWidthOfWirePlane);
+      uparvals.emplace_back(wg.wideWidthOfWirePlane);
+      uparvals.emplace_back(wg.lengthOfWirePlane);
+      uparvals.emplace_back(wg.wiresInEachGroup.size());
+
+      for (CSCWireGroupPackage::Container::const_iterator it = wg.wiresInEachGroup.begin();
+           it != wg.wiresInEachGroup.end();
+           ++it) {
+        uparvals.emplace_back(*it);
+      }
+      for (CSCWireGroupPackage::Container::const_iterator it = wg.consecutiveGroups.begin();
+           it != wg.consecutiveGroups.end();
+           ++it) {
+        uparvals.emplace_back(*it);
+      }
+
+      /** end crap **/
```

Comment:
hmm...?

Reply:
This a comment is taken from the DD side of the code. Maybe is it better to leave it?